### PR TITLE
Fix feature gate RetryGenerateName for v1.30

### PR DESCRIPTION
**Pull request by tengqm**

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
LGTM for docs

/sig api-machinery

File: content/en/docs/reference/command-line-tools-reference/feature-gates/retry-generate-name.md
Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/retry-generate-name.md:1]: We already have the `RetryGenerateName` feature gate [description file (here)](https://github.com/kubernetes/website/blob/main/content/en/docs/reference/command-line-tools-reference/feature-gates/name-generation-retries.md?plain=1), but it’s currently under the wrong file name _'name-generation-retries.md'_.  I suggest we rename the existing file and use its correct content.

Comment (tengqm) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/retry-generate-name.md:1]: That is a problem. We should have named the feature files as the feature name.
There is no rules governing this PascalCase to snake_case translation and
this translation is unnecessary. It only complicates the maintenance of the gates.

Comment (dipesh-rawat) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/retry-generate-name.md:1]: For this particular case, it seems we noticed the discrepancy between the originally documented feature name "NameGenerationRetries" and the actual feature gate name "RetryGenerateName" in the code a few months after the release. Although we later corrected the name in the doc, we missed updating the file name to reflect the change. 

Refer https://github.com/kubernetes/website/commit/91ba3c31746927888c863a5fcf5b12bccd0094be

Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/retry-generate-name.md:1]: Here's what I recommend: add a CI test to tell reviewers when the feature gate name doesn't match up with the file.

The snake case / UpperCamelCase thing isn't relevant to that; even if the filename had used UpperCamelCase, commit https://github.com/kubernetes/website/commit/91ba3c31746927888c863a5fcf5b12bccd0094be could have gone through without the check I'm recommending. If we only hope that people will follow any convention, and don't automate the checks, eventually we'll see this kind of friction.

I also recommend testing to validate that each feature gate is only defined once (in fact I thought I'd implemented that check). 

Why kebab case? Well, almost all our other files use that. It makes the mental model easier for contributors than having to teach which Markdown files use which case convention for their filename.

> rules governing this PascalCase to snake_case translation

- It's usually called “kebab-case“ not snake_case

**Review by sftim — COMMENTED** [review]
This won't change what's published to the live site.

/lgtm
/approve

**Pull request merged**